blocks/identity-block/components/social-sign-on/utils/useSocialSignIn.js

```diff
@@ -54,39 +83,96 @@ function useSocialSignIn(redirectURL, isOIDC, socialSignOnIn, onError = () => {}
 							window.location = validatedURL;
 						}
 					}),
-				auto_select: true,
+				auto_select: false,
+				state_cookie_domain: "none",
 			};
+			// one-tap Google Sign-In
+			window.google.accounts.id.initialize(googleIdConfigOneTap);
 
-			window.google.accounts.id.initialize(googleIdConfig);
-			const googleTextType = socialSignOnIn === SIGN_UP ? 'signup_with' : 'signin_with';
-			window.google.accounts.id.renderButton(document.getElementById("google-sign-in-button"), {
-				type: "standard",
-				theme: "outline",
-				size: "large",
-				text: googleTextType,
-				shape: "rectangular",
-				logo_alignment: "left",
-				width: "400",
-			});
+			if (!customButtons) {
+				const googleTextType = socialSignOnIn === SIGN_UP ? "signup_with" : "signin_with";
+				window.google.accounts.id.renderButton(document.getElementById("google-sign-in-button"), {
+					// render a standard Google Sign-In button
+					type: "standard",
+					theme: "outline",
+					size: "large",
+					text: googleTextType,
+					shape: "rectangular",
+					logo_alignment: "left",
+					width: "400",
+				});
+			}
 
 			Identity.isLoggedIn().then((isLoggedIn) => {
 				if (!isLoggedIn) {
+					// If user rejects it will not appear again based on exponential cooldown https://developers.google.com/identity/gsi/web/guides/features#exponential_cooldown
 					window.google.accounts.id.prompt();
 				} else if (isLoggedIn && isOIDC) {
 					loginByOIDC();
 				}
 			});
 		}
-	}, [config.googleClientId, Identity, isGoogleLoaded, isOIDC, loginByOIDC, redirectURL, socialSignOnIn]);
+	}, [
+		config.googleClientId,
+		Identity,
+		isGoogleLoaded,
+		isOIDC,
+		loginByOIDC,
+		redirectURL,
+		socialSignOnIn,
+		customButtons,
+	]);
 
 	useEffect(() => {
 		const initializeFacebook = async () => {
-			await Identity.initFacebookLogin(null);
+			// Basic setup: https://developers.facebook.com/docs/javascript/quickstart/
+			await Identity.initFacebookLogin(null, null, customButtons);
 		};
 		if (config.facebookAppId) {
 			initializeFacebook();
 		}
-	}, [Identity, config.facebookAppId]);
+	}, [Identity, config.facebookAppId, customButtons]);
+
+	useEffect(() => {
+		const checkIsLoggedIn = () => {
+			setTimeout(() => {
```

Comment:
Whenever setting a timeout inside a useEffect, you should have a return to clear the timeout.  useEffects should have a "cleanup".  The reason for this is that a component can be rerendered by it's parent and if there's no cleanup, this timeout can still exist and may get called twice.

A common pattern for this is similar to the addEventListener I noted above:
```
useEffect (() => {
  const timeout = setTimeout(() => { /*... timeout code ...*/ }, 1500)
  return () => clearTimeout(timeout)
}, [dependency])
```

Without that, if a parent component re-renders, the isLoggedIn function may be called multiple times, the promise returned for each of them, the redirect validated multiple times, and the location changed for each of them.  I would not be surprised if this was flagged by something.  Or even worse, not flagged by something if it's trying to get around a security scan...